packages/lib/server/queries/teams/index.ts

```diff
@@ -120,12 +124,30 @@ export async function getTeamWithMembers(args: {
   }
   if (id) where.id = id;
   if (slug) where.slug = slug;
+  if (isOrgView) {
+    // We must fetch only the organization here.
+    // Note that an organization and a team that doesn't belong to an organization, both have parentId null
+    // If the organization has null slug(but requestedSlug is 'test') and the team also has slug 'test', we can't distinguish them without explicitly checking the metadata.isOrganization
+    // Note that, this isn't possible now to have same requestedSlug as the slug of a team not part of an organization. This is legacy teams handling mostly. But it is still safer to be sure that you are fetching an Organization only in case of isOrgView
+    where.metadata = {
+      path: ["isOrganization"],
+      equals: true,
+    };
+  }
 
-  const team = await prisma.team.findFirst({
+  const teams = await prisma.team.findMany({
     where,
     select: teamSelect,
   });
 
+  if (teams.length > 1) {
```

Comment:
Even though I am ensuring now that there should be one team, it could still happen in future and might be happening right now for some unknown reason.

So, I am logging it as an error but plan to convert it into an actual error, we shouldn't be serving accidentally different team here.